nux/components/dot-tip/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { defer, partial } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createRef, compose } from '@wordpress/element';
+import { Popover, Button, IconButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+export class DotTip extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.popoverRef = createRef();
+	}
+
+	componentDidMount() {
+		if ( this.props.isVisible ) {
+			// Fix the tip not appearing next to the inserter toggle by forcing Popover
+			// to recalculate its size and position on the next frame
+			defer( () => {
```

Comment:
Why is this necessary? do you think this is a Popover bug instead and should be fixed there?

Reply:
The first tip that appears on page load is mounted before layout has finished and this messes up the position calculation. I've moved the fix into Popover where it looks a littles less hacky in 6d8173b2d

Reply:
OK, I dug into this.

The problem is that, during the initial page load, the popover position is calculated while `PostTitle` is hidden because `PostTypeSupportCheck` is waiting for the `getPostType` resolver to complete:

<img width="1112" alt="screen shot 2018-05-31 at 15 25 08" src="https://user-images.githubusercontent.com/612155/40763216-6e1f2828-64e7-11e8-89a6-81ff2363b67f.png">

When the resolver eventually finishes, `PostTitle` appears and the popover is now in the wrong position:

<img width="1111" alt="screen shot 2018-05-31 at 15 30 27" src="https://user-images.githubusercontent.com/612155/40763256-a3ad744a-64e7-11e8-8646-49bbdb6cd75b.png">

To fix this properly, `Popover` needs to recalculate its position whenever its anchor moves (e.g. because of a DOM change).

I'm not sure how best to do that. Two ideas:

1. We could use `MutationObserver` to watch for changes to the DOM. This would be quite expensive though since we'd have to trigger a recalculation whenever **any** of the anchor's parents changes. We could mitigate the performance penalty by throttling the recalculations and/or making it opt-in (e.g. so that only the first `DotTip` does this). 
2. We could recalculate the position e.g. in a `setInterval`. Bad idea—sorry, that's all I've got.

Thoughts? Alternative ideas?

Reply:
Yes, that's not an easy problem to solve. I wonder why `getPostType` is not resolving immediately, I thought we were preloading all the required data on initial load.

cc @aduth

Reply:
Let's leave the workaround as is for now. I've updated the inline comment to better explain the problem and created a issue to follow-up on here: https://github.com/WordPress/gutenberg/issues/7071